Common/Orders/TimeInForces/DayTimeInForceHandler.cs

```diff
@@ -0,0 +1,91 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Interfaces;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Orders.TimeInForces
+{
+    /// <summary>
+    /// Handles the Day time in force for an order (DAY)
+    /// </summary>
+    public class DayTimeInForceHandler : ITimeInForceHandler
+    {
+        private readonly IAlgorithm _algorithm;
+        private readonly MarketHoursDatabase _marketHoursDatabase;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DayTimeInForceHandler"/> class
+        /// </summary>
+        /// <param name="algorithm">The instance of the algorithm</param>
+        /// <param name="marketHoursDatabase">The market hours database</param>
+        public DayTimeInForceHandler(IAlgorithm algorithm, MarketHoursDatabase marketHoursDatabase)
+        {
+            _algorithm = algorithm;
+            _marketHoursDatabase = marketHoursDatabase;
+        }
+
+        /// <summary>
+        /// Checks if an order has expired
+        /// </summary>
+        /// <param name="order">The order to be checked</param>
+        /// <returns>Returns true if the order has expired, false otherwise</returns>
+        public bool HasOrderExpired(Order order)
+        {
+            var exchangeHours = _marketHoursDatabase.GetExchangeHours(order.Symbol.ID.Market, order.Symbol, order.Symbol.SecurityType);
+
+            var orderTime = order.Time.ConvertFromUtc(exchangeHours.TimeZone);
+            var time = _algorithm.UtcTime.ConvertFromUtc(exchangeHours.TimeZone);
```

Comment:
This algorithm dependency seems heavy handed. It looks like we're only using the algorithm to get the current time. A common pattern in lean models is to accept the `Security` object in the method. Accepting the security object would remove the algorithm dependency *and* the market hours database dependency, since the current time and the exchange hours can both be directly resolved form the security object.